src/CMakeLists.txt

```diff
@@ -8,6 +8,22 @@
 # SPDX-License-Identifier: BSD-3-Clause
 #
 
+# Generate version header
+find_program(GIT_EXE NAMES git)
+string(TIMESTAMP DLAF_TIMESTAMP "%Y-%m-%d %H:%M:%S")
+if(DEFINED GIT_EXE AND EXISTS "${PROJECT_SOURCE_DIR}/.git")
+  execute_process(COMMAND git rev-parse HEAD
+                  OUTPUT_VARIABLE DLAF_SHA
+                  WORKING_DIRECTORY "${PROJECT_SOURCE_DIR}"
+                  ERROR_QUIET
+                  OUTPUT_STRIP_TRAILING_WHITESPACE)
+else()
+  set(SIRIUS_SHA "N/A : archive")
```

Comment:
```suggestion
  set(DLAF_SHA "N/A : archive")
```

Reply:
Good catch, this PR is based on work I did for header generation in SIRIUS.